proto/api/v1/network_graph_service.proto

```diff
@@ -73,11 +74,24 @@ message GetExternalNetworkEntitiesResponse {
 
 message GetExternalNetworkFlowsRequest {
   string cluster_id = 1;
-  string deployment_id = 2;
+  string entity_id = 2;
+  string query = 3;
+  google.protobuf.Timestamp since = 4;
+  Pagination pagination = 5;
 }
 
 message GetExternalNetworkFlowsResponse {
-  repeated storage.NetworkFlow flows = 1;
+  storage.NetworkEntityInfo entity = 1;
+  repeated storage.NetworkFlow flows = 2;
+}
+
+message ExternalNetworkFlowMetadata {
+  storage.NetworkEntityInfo entity = 1;
+  int32 flowsCount = 2;
+}
+
+message GetExternalNetworkFlowsMetadataResponse {
+  repeated ExternalNetworkFlowMetadata data = 1;
```

Comment:
Not a change request, but pointing out that naming this data feels a little awkward. Grabbing what we need ends up looking something like this -> `response.data.data`

Reply:
Ah yeah agreed, that's not ideal. I've switched it to "entities", hopefully that's a bit better